be/src/util/blocking_priority_queue.hpp

```diff
@@ -154,14 +157,15 @@ class BlockingPriorityQueue {
         }
 
         _queue.push(val);
+        unique_lock.unlock();
```

Comment:
why add lock for notify?? https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one

Reply:
putting the notify_one before unlock:
notifying thread: notify -> eventually release lock
notified thread: awaken -> attempt to acquire lock and fail -> block until lock available -> acquire lock after 
notifying thread releases it

putting the notify_one after unlock:
notifying thread: notify 
notified thread: awaken -> attempt to acquire lock and succeed

https://stackoverflow.com/questions/55154462/c-condition-variable-notify-one-before-or-after-release-lock